### PR TITLE
Graphite emitter - add plaintext protocol

### PR DESCRIPTION
**Pull request by blugowski**
This patch adds two configuration options to graphite-emitter extension:
- `protocol` - with addtional plaintext protocol, up to now there was only pickle
- `replaceSlashWithDot` - posibility of sending metrics without slashes by replacing them with dots(#3537)

ex.
Now metrics structure looks like this:
```
druid.broker.query/cache/delta/hit
             query/cache/delta/miss
```

After:
```
druid.broker.query.cache.delta.
                               hit
                               miss
```

### CODE REVIEW

File: docs/content/development/extensions-contrib/graphite.md [hunk 1]
```diff
@@ -52,10 +53,15 @@ The path will be in the form `<namespacePrefix>.[<druid service name>].[<druid h
...
 
 You can omit the hostname by setting `ignoreHostname=true`
-`druid.SERVICE_NAME.dataSourceName.queryType.query.time`
+`druid.SERVICE_NAME.dataSourceName.queryType.query/time`
 
 You can omit the service name by setting `ignoreServiceName=true`
-`druid.HOSTNAME.dataSourceName.queryType.query.time`
+`druid.HOSTNAME.dataSourceName.queryType.query/time`
+
+Elements in metric name by default are separated by "/", so graphite will create all metrics on one level. If you want to have metrics in the tree structure, you have to set `replaceStashWithDot=true`
```
Comment (b-slim) [inline docs/content/development/extensions-contrib/graphite.md]: slash instead of stash

Comment (b-slim) [inline docs/content/development/extensions-contrib/graphite.md]: how about adding the entire property name `druid.x.x.replaceSlashWithDot`

**Comment by b-slim** [comment]
@blugowski thanks for the contribution, i am on board with the change but if my understanding is correct the plain text one you are adding will not support batching of multiple events. If my claim is correct we need to make sure to update the docs to make sure that users understand the eventual difference in terms of operation and performance as well.

File: docs/content/development/extensions-contrib/graphite.md (hunk 1 above)
Comment (blugowski) [inline docs/content/development/extensions-contrib/graphite.md]: Fixed.

Comment (blugowski) [inline docs/content/development/extensions-contrib/graphite.md]: `replaceSlashWithDot` is configuration option for converters, same as `ignoreHostname` and `ignoreServiceName`.

**Comment by blugowski** [comment]
@b-slim yes, you're correct, batching is supported only by pickle protocol. I've added appropriate comments in _Introduction_ and description of `druid.emitter.graphite.batchSize`. Is it good enough?

File: docs/content/development/extensions-contrib/graphite.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ To use this extension, make sure to [include](../../operations/including-extensi
 ## Introduction
 
 This extension emits druid metrics to a graphite carbon server.
-Events are sent after been [pickled](http://graphite.readthedocs.org/en/latest/feeding-carbon.html#the-pickle-protocol); the size of the batch is configurable. 
+Metrics can be sent by using [plaintext](http://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-plaintext-protocol) or [pickle](http://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-pickle-protocol) protocol; the size of the batch is configurable(only pickle protocol).
```
Comment (b-slim) [inline docs/content/development/extensions-contrib/graphite.md]: nit space before `(`. Can we add explanation about the difference between plain text and pickled one, like the performance implication ? i think we need to state that in the plaintext is sending an event per connection which can be very inefficient.

File: docs/content/development/extensions-contrib/graphite.md [hunk 3]
```diff
@@ -19,7 +19,8 @@ All the configuration parameters for graphite emitter are under `druid.emitter.g
 |--------|-----------|---------|-------|
 |`druid.emitter.graphite.hostname`|The hostname of the graphite server.|yes|none|
 |`druid.emitter.graphite.port`|The port of the graphite server.|yes|none|
-|`druid.emitter.graphite.batchSize`|Number of events to send as one batch.|no|100|
+|`druid.emitter.graphite.batchSize`|Number of events to send as one batch(only for pickle protocol)|no|100|
```
Comment (b-slim) [inline docs/content/development/extensions-contrib/graphite.md]: space before `(`

Comment (blugowski) [inline docs/content/development/extensions-contrib/graphite.md]: Fixed.

File: docs/content/development/extensions-contrib/graphite.md (hunk 2 above)
Comment (blugowski) [inline docs/content/development/extensions-contrib/graphite.md]: Fixed.

**Comment by b-slim** [comment]
@blugowski can fill the CLA http://druid.io/community/cla.html

**Comment by blugowski** [comment]
@b-slim done.

**Comment by b-slim** [comment]
👍

**Comment by LoicAG** [comment]
Thank you for this PR, both of these features would be very useful for our use case; in which release will it be included?

**Comment by b-slim** [comment]
will run test one more time and merge it afterwords, expect this to be in the next release.

**Comment by b-slim** [comment]
@blugowski can you please fix the style issues ?

**Comment by b-slim** [comment]
```
[ERROR] /home/travis/build/druid-io/druid/extensions-contrib/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEmitter.java:143:31: '{' at column 31 should be on a new line. [LeftCurly]
````

**Comment by blugowski** [comment]
@b-slim fixed.

**Pull request merged**